crates/npmrc/src/custom_deserializer.rs

```diff
@@ -22,19 +22,19 @@ pub fn default_public_hoist_pattern() -> Vec<String> {
 /// On macOS: ~/Library/pacquet/store
 /// On Linux: ~/.local/share/pacquet/store
 pub fn default_store_dir() -> PathBuf {
-    // TODO: If env variables start with ~, make sure to resolve it into home_dir.
+    let home_dir = dirs::home_dir().expect("Home directory is not available");
+
     if let Ok(pacquet_home) = env::var("PACQUET_HOME") {
-        return PathBuf::from(pacquet_home).join("store");
+        // if variable start with `~/`,it should be removed,or use default value.
+        let pacquet_home = pacquet_home.strip_prefix("~/").unwrap_or(pacquet_home.as_str());
+        return home_dir.join(pacquet_home).join("store");
```

Comment:
In pnpm, if one specify `PNPM_HOME` to be `pnpm-home`, pnpm will use `$(pwd)/pnpm-home` as home. Not `~/pnpm-home`. The pacquet implementation should mirror that of pnpm.